### PR TITLE
Add Accept-Language Header

### PR DESCRIPTION
**Pull request by lbalmaceda**
Applies to all AuthenticationAPIClient calls. If the language is not found/available, then it will default to "en".

### CODE REVIEW

File: auth0/src/main/java/com/auth0/android/request/internal/RequestFactory.java [hunk 1]
```diff
@@ -137,4 +140,9 @@ AuthenticationRequest createAuthenticationRequest(HttpUrl url, OkHttpClient clie
     Map<String, String> getHeaders() {
         return headers;
     }
+
+    static String getDefaultLocale() {
+        String language = Locale.getDefault().toString();
+        return language != null ? language : "en";
```
Comment (nikolaseu) [inline auth0/src/main/java/com/auth0/android/request/internal/RequestFactory.java]: I think we should check for empty string, according to the docs it says an empty string is returned in case there's no lang or country.

File: auth0/src/test/java/com/auth0/android/authentication/AuthenticationAPIClientTest.java [hunk 2]
```diff
@@ -1283,4 +1339,9 @@ public void shouldParseUnauthorizedPKCEError() throws Exception {
         }.getType();
         return gson.fromJson(request.getBody().readUtf8(), mapType);
     }
+
+    private String getDefaultLocale() {
```
Comment (nikolaseu) [inline auth0/src/test/java/com/auth0/android/authentication/AuthenticationAPIClientTest.java]: wouldn't it be better if we add a test for the actual function  https://github.com/auth0/Auth0.Android/pull/15/files#diff-f8d8baa1565ca2a3ec49d45e48f3febdR144
and then use that same one?

File: auth0/src/test/java/com/auth0/android/request/internal/RequestFactoryTest.java [hunk 3]
```diff
@@ -58,11 +58,19 @@ public void shouldHaveNonNullHeaders() throws Exception {
     }
 
     @Test
+    public void shouldHaveAcceptLanguageHeader() throws Exception {
```
Comment (nikolaseu) [inline auth0/src/test/java/com/auth0/android/request/internal/RequestFactoryTest.java]: is this used?

File: auth0/src/test/java/com/auth0/android/authentication/AuthenticationAPIClientTest.java (hunk 2 above)
Comment (lbalmaceda) [inline auth0/src/test/java/com/auth0/android/authentication/AuthenticationAPIClientTest.java]: If you mean to test `getDefaultLocale()` then that's not possible with Mockito, as the `Locale` class has the _static_ & _final_ modifiers and can't be mocked.

**Comment by nikolaseu** [comment]
LGTM

File: auth0/src/main/java/com/auth0/android/request/internal/RequestFactory.java [hunk 4]
```diff
@@ -137,4 +140,9 @@ AuthenticationRequest createAuthenticationRequest(HttpUrl url, OkHttpClient clie
     Map<String, String> getHeaders() {
         return headers;
     }
+
+    static String getDefaultLocale() {
+        String language = Locale.getDefault().toString();
+        return !language.isEmpty() ? language : "en";
```
Comment (hzalaz) [inline auth0/src/main/java/com/auth0/android/request/internal/RequestFactory.java]: `en` is not a Locale

Comment (lbalmaceda) [inline auth0/src/main/java/com/auth0/android/request/internal/RequestFactory.java]: I took it from here https://gist.github.com/jasef/337431c43c3addb2cbd5eb215b376179 but I guess I can add the country. Defaulting to US seems fine, anyway this is an edge case. The phone should have a valid locale.

File: auth0/src/test/java/com/auth0/android/authentication/AuthenticationAPIClientTest.java (hunk 2 above)
Comment (lbalmaceda) [inline auth0/src/test/java/com/auth0/android/authentication/AuthenticationAPIClientTest.java]: done !
